pkg/controller/sdk/reconciler.go

```diff
@@ -51,9 +51,9 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 
 	logger.Infof("Reconciling %s %v", r.provider.Parent.GetObjectKind(), request)
 
-	obj := r.provider.Parent.DeepCopyObject()
+	original := r.provider.Parent.DeepCopyObject()
```

Comment:
This was the bug @grantr 

I was passing in the cache's version of the object by mistake. And then assigning the returned value back to the pointer but those point to the caches version all the time....